### ISSUE TITLE
Custom adventure-mode dimensions for boss battles and exploration

### ISSUE THREAD

**Issue by GreatestFool** [GreatestFool/crystialis#15]
Some dungeons will likely have to be custom made from the ground up in static void dimensions that follow the pack in some fashion.

[Dimensional Dungeons](https://www.curseforge.com/minecraft/mc-mods/dimensional-dungeons) is another way to do this, and will likely be used, considering that since issue 11 on [this repository](https://github.com/AllenSeitz/DimDungeons/).

I remember seeing a thread about this once, regarding a pack where all bosses could only be fought inside of custom adventure-mode dungeons, but haven't been able to find it. Regardless, I vaguely remember how it worked. You use two (or more) command blocks, and set one to set you to adventure gamemode within a specified range while the other sets you to survival gamemode when you leave the range. The second one worked either by using the same range, or a little buffer right after the adventure gamemode area, to detect the player in and switch their gamemode around.

This will require either custom built dungeons, or the modification of existing dungeons should they support it.

**Comment by GreatestFool** [comment]
Currently, there's an open issue (101) on [Portality's GitHub](https://github.com/InnovativeOnlineIndustries/Portality), not linking directly because it'll appear in the issue log, and this thread is out of scope for that mod.

Regardless, currently, vertical portals are not possible. For the boss room, as currently envisioned, to be possible. Either this needs to be added or some other solution will have to be found.

Will investigate.

**Comment by GreatestFool** [comment]
Right, a [solution has been found](https://www.curseforge.com/minecraft/mc-mods/wormhole-portals) at it's even a superior one! Probably, at least. I've raised an issue over on [Wormhole](https://github.com/SuperMartijn642/Wormhole)'s github, number 20 specifically, for redstone control. Not linking directly, because this is out of scope.

Regardless, while the portals currently do not support redstone portal activation on/off state toggling, it does support vertical and horizontal portals. Not only that, but they do not actually link portals together, instead you add a target defined through a controller to a portal and teleport to a place in a three-dimensional-space. This supports multi-dimensional portals as well!

Basically, this means you can have one-way portals. Which is great for my use case. One portal leads into the boss room, and another leads back to the hallway!
